apps/api-portal/features/user-applications/user-applications.ts

```diff
@@ -0,0 +1,109 @@
+import { stringify } from 'qs'
+import { QueryClient, useMutation, useQuery, useQueryClient } from 'react-query'
+import { useMemo, useState, useEffect } from 'react'
+import { FieldValidationError } from 'lib/types'
+import { GFWAPI } from '@globalfishingwatch/api-client'
+import { UserApplication } from '@globalfishingwatch/api-types'
+import useUser, { checkUserApplicationPermission } from 'features/user/user'
+
+export type UserApplicationCreateArguments = Omit<UserApplication, 'id' | 'token' | 'createdAt'>
+export interface UserApplicationFetchResponse {
+  offset: number
+  metadata: any
+  total: number
+  limit: number | null
+  nextOffset: number
+  entries: UserApplication[]
+}
+
+async function fetchUserApplications(userId: number, limit = 0, offset = 0) {
+  const query = stringify({
+    'user-id': userId,
+    ...((limit && { limit }) || {}),
+    ...((offset && { offset }) || {}),
+  })
+  const url = `/v2/auth/user-applications?${query}`
+  const data = await GFWAPI.fetch<UserApplicationFetchResponse>(url).catch((error) => {
+    return null
+  })
+  return data
+}
+
+export const useUserApplications = (userId) =>
+  useQuery(['user-applications', userId], () => fetchUserApplications(userId), {})
+
+async function deleteUserApplication(id: number) {
+  const url = `/v2/auth/user-applications/${id}`
+  const userApplication = await GFWAPI.fetch<UserApplication>(url, {
+    method: 'DELETE',
+    responseType: 'default',
+  })
+  return { ...userApplication, id }
+}
+
+export const useDeleteUserApplication = (userId) => {
```

Comment:
what about decoupling this from the component to its own hook file? 
I think that is easier to reuse and to read